lua/nvim-treesitter/ts_utils.lua

```diff
@@ -212,4 +212,52 @@ function M.get_node_at_cursor(winnr)
   return root:named_descendant_for_range(cursor[1]-1,cursor[2],cursor[1]-1,cursor[2])
 end
 
+
+-- Set visual selection to node
+function M.update_selection(buf, node)
+  local start_row, start_col, end_row, end_col = node:range()
+
+  if end_row == vim.fn.line('$') then
+    end_col = #vim.fn.getline('$')
+  end
+
+  vim.fn.setpos(".", { buf, start_row + 1, start_col + 1, 0 })
+  vim.fn.nvim_exec("normal v", false)
+
+  -- Convert exclusive end position to inclusive
+  if end_col == 0 then
+    vim.fn.setpos(".", { buf, end_row, -1, 0 })
```

Comment:
why is `end_row` not changed here but changed in the `else` part ?

Reply:
This condition code for two things:
- convert from 1-based indices to 0-based indices
- subtract one character, this changes the row of it's the first character

Reply:
oh i see, my bad.

Reply:
But I should simplify this part to make it obvious